From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ROCKNIX <rocknix@rocknix.org>
Date: Sat, 07 Mar 2026 00:00:00 +0000
Subject: [PATCH] arm64: dts: rockchip: px30: add DMC interrupt, OPPs, disable by default

Add GIC_SPI 105 completion_irq to DMC node and expand the DDR OPP
table from single 924 MHz to 6 BSP frequency points (194-924 MHz).

Set both DFI and DMC nodes to status = "disabled" by default.
Per-device DTS overlays or board files enable them after verifying
ATF firmware support.  The px30_dmc driver (module) will only probe
when the node is enabled.

Signed-off-by: ROCKNIX <rocknix@rocknix.org>
---
--- a/arch/arm64/boot/dts/rockchip/px30.dtsi
+++ b/arch/arm64/boot/dts/rockchip/px30.dtsi
@@ -343,22 +343,44 @@
 		reg = <0x00 0xff610000 0x00 0x400>;
 		compatible = "rockchip,px30-dfi";
 		rockchip,pmu = <&pmugrf>;
-		status = "okay";
+		status = "disabled";
 	};
 
 	dmc: dmc {
 		compatible = "rockchip,px30-dmc";
+		interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "complete_irq";
 		devfreq-events = <&dfi>;
 		clocks = <&cru SCLK_DDRCLK>;
 		clock-names = "dmc_clk";
 		operating-points-v2 = <&dmc_opp_table>;
 		uptreshold = <40>;
 		downdifferential = <20>;
-		status = "okay";
+		status = "disabled";
 	};
 
 	dmc_opp_table: dmc_opp_table {
 		compatible = "operating-points-v2";
+		opp-194000000 {
+			opp-hz = /bits/ 64 <194000000>;
+			opp-microvolt = <950000>;
+		};
+		opp-328000000 {
+			opp-hz = /bits/ 64 <328000000>;
+			opp-microvolt = <950000>;
+		};
+		opp-450000000 {
+			opp-hz = /bits/ 64 <450000000>;
+			opp-microvolt = <950000>;
+		};
+		opp-666000000 {
+			opp-hz = /bits/ 64 <666000000>;
+			opp-microvolt = <1050000>;
+		};
+		opp-786000000 {
+			opp-hz = /bits/ 64 <786000000>;
+			opp-microvolt = <1100000>;
+		};
 		opp-924000000 {
 			opp-hz = /bits/ 64 <924000000>;
 			opp-microvolt = <1125000>;
